secboot/secboot_tpm.go

```diff
@@ -277,14 +283,54 @@ func newAuthRequestor() (sb.AuthRequestor, error) {
 	)
 }
 
+// TODO: consider moving this to secboot
+func readKeyFileImpl(keyfile string) (*sb.KeyData, *sb_tpm2.SealedKeyObject, error) {
+	f, err := os.Open(keyfile)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer f.Close()
+
+	var rawPrefix = []byte("USK$")
+
+	buf := make([]byte, len(rawPrefix))
+	if _, err := io.ReadFull(f, buf); err != nil {
+		return nil, nil, err
+	}
+	if bytes.HasPrefix(buf, rawPrefix) {
+		sealedObject, err := sbReadSealedKeyObjectFromFile(keyfile)
+		if err != nil {
+			return nil, nil, fmt.Errorf("cannot read key object: %v", err)
+		}
+		keyData, err := sbNewKeyDataFromSealedKeyObjectFile(keyfile)
+		if err != nil {
+			return nil, nil, fmt.Errorf("cannot read key object as key data: %v", err)
+		}
+		return keyData, sealedObject, err
+
+	} else {
```

Comment:
new path doesn't seem to be tested, same for other functions here

Reply:
I am changing and adding tests to reach those. But it has to be noted that putting new key data format into the key files is only a temporary thing that we will remove in future PRs when we put the key data in the key slots. So the tests will have to be different again. With maybe some reverts.